zipkin-storage/cassandra-v1/src/main/java/zipkin2/storage/cassandra/v1/IndexTraceId.java

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright 2015-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin2.storage.cassandra.v1;
+
+import com.datastax.driver.core.BoundStatement;
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.ResultSetFuture;
+import com.datastax.driver.core.Session;
+import com.datastax.driver.core.querybuilder.Insert;
+import com.datastax.driver.core.querybuilder.QueryBuilder;
+import com.google.auto.value.AutoValue;
+import com.google.common.cache.CacheBuilder;
+import java.util.AbstractMap.SimpleImmutableEntry;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import zipkin2.storage.QueryRequest;
+import zipkin2.storage.cassandra.internal.call.DeduplicatingVoidCallFactory;
+import zipkin2.storage.cassandra.internal.call.ResultSetFutureCall;
+
+/**
+ * Inserts index rows into a Cassandra table. This skips entries that don't improve results based on
+ * {@link QueryRequest#endTs()} and {@link QueryRequest#lookback()}. For example, it doesn't insert
+ * rows that only vary on timestamp and exist between timestamps of existing rows.
+ */
+final class IndexTraceId extends ResultSetFutureCall<Void> {
+  static final Logger LOG = LoggerFactory.getLogger(IndexTraceId.class);
+
+  @AutoValue
+  abstract static class Input {
+    static Input create(String partitionKey, long timestamp, long traceId) {
+      return new AutoValue_IndexTraceId_Input(partitionKey, timestamp, traceId);
+    }
+
+    abstract String partitionKey(); // ends up as a partition key, ignoring bucketing
+
+    abstract long ts(); // microseconds at millis precision
+
+    abstract long trace_id(); // clustering key
```

Comment:
Ditto, do we usually name our methods like this even for json?

Reply:
similar to above, these match the input variables in CQL. it isn't about json really, and is all over the cassandra v1 codebase.